luigi_pipeline/lib/hail_tasks.py

```diff
@@ -41,7 +41,8 @@ class HailMatrixTableTask(luigi.Task):
                                                                                             'to annotate vep.')
 
     def requires(self):
-        return [VcfFile(filename=s) for s in self.source_paths]
+        # We need to exclude globs in source path since they aren't files.
+        return [VcfFile(filename=s) for s in self.source_paths if '*' not in s]
```

Comment:
When does a glob get passed as a source path? If globs won't work, it seems better to raise an error or let the task fail by creating the VcfFile instead of silently ignoring them.

Reply:
Globs do work, the requires is used to check whether the files exist. So the param is source_paths and we only check that non globs exist, but globs are still used later down.
Sorry the comment was mine, it's probably confusing. @mike-w-wilson can you change it to specify that it's excluded only for checking that it exists?

Reply:
Some callsets get delivered sharded such as 

> gs://seqr-datasets/GRCh38/RDG_WGS_Broad_Internal/v6/sharded_vcf/RDG_WGS_Broad_Internal_v5.filtered.*.vcf.gz 

Where the glob the shard number. @Kevin what do you think of Nick's suggestion?

Reply:
@mike-w-wilson I think @nawatts misunderstood that the globs were not being used. We changed https://github.com/macarthur-lab/hail-elasticsearch-pipelines/pull/147/files/f693d335c21b552eafd9595f6057312f0ce6eb2a#diff-9e4438316786f665992553279456402cL57 here instead of using `input()` (which is derived from `requires()` to go straight to the source_paths which has the globs.
Thus, the logic is source_paths can have globs, but they will not be in the requires because they cannot be checked to exist. I mean, if we're sure *'s are only used at the file level, we can extract the directory and make sure that it exists, but I'm not sure if *'s can be used in dirs as well.

Reply:
I see. Updated the comment.